File: yunhwan/프로그래머스 가장 큰 수.py
def solution(numbers):
    answer = ''
    #문자열로 변환
    arr= list(map(str,numbers))
    
		#1000까지의 수 이므로 비교를 위해 3번을곱해줌
    arr.sort(reverse=True, key=lambda x:x*3)
		#하나의 문자열로 합침
    result =''.join(arr)
		#001 같은 경우를 없애기 위해서 숫자로 변환 후 다시 문자로 변환
    answer = str(int(result))
    return answer

File: jiWoo/프로그래머스/프로그래머스_가장큰수.java
package jiWoo.프로그래머스;
import java.util.*;
public class 프로그래머스_가장큰수 {
    public String solution(int[] numbers) {
        String answer = "";
        String numString[] =new String[numbers.length];
        for(int i=0;i<numbers.length;i++){
            numString[i]=Integer.toString(numbers[i]);
        }
        Arrays.sort(numString,new Comparator<>(){
            @Override
            public int compare(String o1, String o2){
                return (o2+o1).compareTo(o1+o2);
            }
        });
        answer=String.join(answer,numString);
        // 숫자가 다 0일 때
        if(numString[0].equals("0")){
            answer="0";
        }
        return answer;
    }
}
